py_experimenter/experimenter.py

```diff
@@ -29,58 +29,101 @@ def __init__(self,
-            self._config.set('PY_EXPERIMENTER', 'database', database_name)
+            self.config.set('PY_EXPERIMENTER', 'database', database_name)
         self.name= name
 
-        self._config_file = config_file
-        self.timestamp_on_result_fields = utils.timestamps_for_result_fields(self._config)
+        self.config_file = config_file
+        self.timestamp_on_result_fields = utils.timestamps_for_result_fields(self.config)
 
-        if self._config['PY_EXPERIMENTER']['provider'] == 'sqlite':
-            self._dbconnector = DatabaseConnectorLITE(self._config)
-        elif self._config['PY_EXPERIMENTER']['provider'] == 'mysql':
-            self._dbconnector = DatabaseConnectorMYSQL(self._config, credential_path)
+        if self.config['PY_EXPERIMENTER']['provider'] == 'sqlite':
+            self.dbconnector = DatabaseConnectorLITE(self.config)
+        elif self.config['PY_EXPERIMENTER']['provider'] == 'mysql':
+            self.dbconnector = DatabaseConnectorMYSQL(self.config, credential_path)
         else:
             raise ValueError('The provider indicated in the config file is not supported')
 
         logging.info('Initialized and connected to database')
 
     def set_config_value(self, section_name: str, key: str, value: str) -> None:
         """
-        Sets a value in the configuration file. If the section does not exist, it is created.
-        :param section_name: Name of the section in the configuration file.
-        :param key: Name of the key in the configuration file.
-        :param value: Value to set.
+        Modifies the experiment configuration so that within the given `section_name` the value of the given `key` is overridden, 
+        or created if it is not existing beforehand.
+
+        :param section_name: The name of the section of the experiment configuration in which a value should be set.
+        :type section_name: str
+        :param key: The name of the key within the given section of the experiment configuration in which a value should be set.
+        :type key: str
+        :param value: The value of the key within the given section of the experiment configuration that should be set.
+        :type value: str
         """
-        if not self._config.has_section(section_name):
-            self._config.add_section(section_name)
-        self._config.set(section_name, key, value)
-        PyExperimenter._valid_configuration(self._config, self._credential_path)
+        if not self.config.has_section(section_name):
+            self.config.add_section(section_name)
+        self.config.set(section_name, key, value)
+        PyExperimenter._is_valid_configuration(self.config, self.credential_path)
 
     def get_config_value(self, section_name: str, key: str) -> str:
-        return self._config.get(section_name, key)
+        """
+        Returns the value of the experiment configuration for the given `key`. If the `key` is not contained within the section, 
```

Comment:
Returns the value of the property of the experiment configuration identified by the given key....